embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/session/orchestrator/OrchestratorBoundaryDelegate.kt

```diff
@@ -0,0 +1,38 @@
+package io.embrace.android.embracesdk.session.orchestrator
+
+import io.embrace.android.embracesdk.capture.user.UserService
+import io.embrace.android.embracesdk.logging.InternalErrorService
+import io.embrace.android.embracesdk.ndk.NdkService
+import io.embrace.android.embracesdk.session.MemoryCleanerService
+import io.embrace.android.embracesdk.session.properties.EmbraceSessionProperties
+
+/**
+ * This class is responsible for orchestrating a clean boundary between session envelopes.
+ * I.e. if a session transitions to a background activity, previous data should be cleared
+ * & the relevant services should be updated as necessary.
+ *
+ * This class acts as a delegate to the SessionOrchestrator & is separated out because it
+ * contains references to various services that are otherwise irrelevant to the SessionOrchestrator.
+ */
+internal class OrchestratorBoundaryDelegate(
```

Comment:
Once this is in, I'll move the session span creation to here as well. Maybe after I spin that out of the SpansServiceImpl